### PR TITLE
test(dev-infra): add testing for integration of ReleaseNotes in publishing tooling

### PR DESCRIPTION
**Pull request by josephperrott**
Add testing of the ReleaseNotes integration with the release publishing tooling.

### CODE REVIEW

File: dev-infra/release/publish/release-notes/release-notes.ts [hunk 1]
```diff
@@ -27,19 +28,21 @@ export class ReleaseNotes {
   /** Construct a release note generation instance. */
   static async fromLatestTagToHead(version: semver.SemVer, config: ReleaseConfig):
       Promise<ReleaseNotes> {
-    return new ReleaseNotes(version, config);
+    const instance = new ReleaseNotes(version, config);
+    instance.commits = await getCommitsInRange(instance.git.getLatestSemverTag().format(), 'HEAD');
```
Comment (devversion) [inline dev-infra/release/publish/release-notes/release-notes.ts:32]: Should we rather pass the commits through the constructor? that would make it more obvious that commits _should_ be determined ourselves.

**Review by devversion — APPROVED** [review]
LGTM. one comment about how the commits could be passed through.

File: dev-infra/release/publish/release-notes/release-notes.ts (hunk 1 above)
Comment (josephperrott) [reply dev-infra/release/publish/release-notes/release-notes.ts:32]: Would it make more sense to have the `startingSha` and `endingSha` optionally provided?  That way we are saying these are the release notes that are created for this range of commits and its reproducible?

Comment (devversion) [reply dev-infra/release/publish/release-notes/release-notes.ts:32]: Yeah, that would also work for me 👍

**Review by devversion — APPROVED** [review]

**Pull request closed without merging**